### PR TITLE
Browser setup fix few potential issues. 

### PR DESCRIPTION
**Pull request by guruswarupa**
## Type of Change
- [x] Bug fix

## Description
Browser setup fix few potential issues. 

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/browser-setup.sh [hunk 1]
```diff
@@ -113,87 +123,106 @@ Signed-By: /usr/share/keyrings/librewolf.gpg" | "$ESCALATION_TOOL" tee /etc/apt/
...
-            printf "%b\n" "${RED}The script does not support your Distro. Install manually..${RC}"
-            ;;
-    esac
+    if ! command_exists lynx; then
+        case "$PACKAGER" in
+            apt-get|nala)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y lynx
+                ;;
+            zypper)
+                "$ESCALATION_TOOL" "$PACKAGER" install lynx
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:211]:
```suggestion
                "$ESCALATION_TOOL" "$PACKAGER" install -y lynx
```

File: core/tabs/applications-setup/browser-setup.sh [hunk 2]
```diff
@@ -4,81 +4,91 @@
...
+    else
+        printf "%b\n" "${GREEN}Thorium Browser is already installed.${RC}"
+    fi
+}
 
+install_firefox() {
+    printf "%b\n" "${YELLOW}Installing Mozilla Firefox...${RC}"
+    if ! command_exists firefox ; then
+        case "$PACKAGER" in
+            apt-get)
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:67]:
```suggestion
            apt-get|nala)
```

File: core/tabs/applications-setup/browser-setup.sh [hunk 3]
```diff
@@ -4,81 +4,91 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" update
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:68]:
```suggestion

```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/browser-setup.sh [hunk 4]
```diff
@@ -113,87 +123,106 @@ Signed-By: /usr/share/keyrings/librewolf.gpg" | "$ESCALATION_TOOL" tee /etc/apt/
...
 
 install_vivaldi() {
-    printf "%b\n" "${YELLOW}Installing Vivaldi...${RC}"
-    wget https://downloads.vivaldi.com/snapshot/install-vivaldi.sh
-    sh install-vivaldi.sh
+    if ! command_exists vivaldi; then
+        printf "%b\n" "${YELLOW}Installing Vivaldi...${RC}"
+        wget https://downloads.vivaldi.com/snapshot/install-vivaldi.sh
+        sh install-vivaldi.sh
+        printf "%b\n" "${GREEN}Vivaldi installed successfully!${RC}"
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:170]:
```suggestion
        curl -fsSL https://downloads.vivaldi.com/snapshot/install-vivaldi.sh | sh
        if [ $? -eq 0 ]; then
                printf "%b\n" "${GREEN}Vivaldi installed successfully!${RC}"
        else
                printf "%b\n" "${RED}Vivaldi installation failed!${RC}"
        fi
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/browser-setup.sh [hunk 5]
```diff
@@ -113,87 +122,109 @@ Signed-By: /usr/share/keyrings/librewolf.gpg" | "$ESCALATION_TOOL" tee /etc/apt/
...
+                "$ESCALATION_TOOL" "$PACKAGER" install -y lynx
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm lynx
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install lynx
+                ;;
+            *)
+                printf "%b\n" "${RED}The script does not support your Distro. Install manually..${RC}"
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:222]:
```suggestion
            pacman)
                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm lynx
                ;;
            *)
                "$ESCALATION_TOOL" "$PACKAGER" install -y lynx
                ;;
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

**Comment by guruswarupa** [comment]
> use curl instead of wget

have changed the ones which are used by me as wget to curl -O. the ones that were suggested from the browser installation command isn't modified.

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]
you cannot use wget, use curl...

**Comment by nnyyxxxx** [comment]
> > use curl instead of wget
> 
> have changed the ones which are used by me as wget to curl -O. the ones that were suggested from the browser installation command isn't modified.

use curl

File: core/tabs/applications-setup/browser-setup.sh [hunk 6]
```diff
@@ -4,81 +4,90 @@
...
-
-install_firefox() {
-    printf "%b\n" "${YELLOW}Installing Mozilla Firefox...${RC}"
-    case "$PACKAGER" in
-        apt-get)
+    if ! command_exists thorium-browser; then
+        case "$PACKAGER" in
+        apt-get|nala)
+            "$ESCALATION_TOOL" rm -fv /etc/apt/sources.list.d/thorium.list
+                "$ESCALATION_TOOL" wget --no-hsts -P /etc/apt/sources.list.d/ http://dl.thorium.rocks/debian/dists/stable/thorium.list
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/browser-setup.sh:41]: formatting issue

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]
fix shell check errors

File: core/tabs/applications-setup/browser-setup.sh [hunk 7]
```diff
@@ -4,81 +4,90 @@
...
-            "$ESCALATION_TOOL" "$PACKAGER" install fedora-workstation-repositories
-            "$ESCALATION_TOOL" "$PACKAGER" config-manager --set-enabled google-chrome
-            "$ESCALATION_TOOL" "$PACKAGER" install google-chrome-stable
-            ;;
-        *)
-            printf "%b\n" "${RED}The script does not support your Distro. Install manually..${RC}"
-            ;;
-    esac
-
+    if ! command_exists google-chrome ; then
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:7]:
```suggestion
    if ! command_exists google-chrome; then
```

File: core/tabs/applications-setup/browser-setup.sh [hunk 8]
```diff
@@ -4,81 +4,90 @@
...
-    esac
+        esac
+    else
+        printf "%b\n" "${GREEN}Thorium Browser is already installed.${RC}"
+    fi
+}
 
+install_firefox() {
+    printf "%b\n" "${YELLOW}Installing Mozilla Firefox...${RC}"
+    if ! command_exists firefox ; then
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:65]:
```suggestion
    if ! command_exists firefox; then
```

File: core/tabs/applications-setup/browser-setup.sh [hunk 9]
```diff
@@ -113,87 +122,101 @@ Signed-By: /usr/share/keyrings/librewolf.gpg" | "$ESCALATION_TOOL" tee /etc/apt/
...
-    esac
+    if ! command_exists lynx; then
+        case "$PACKAGER" in
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm lynx
+                ;;
+            *)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y lynx
+                ;;
+                ;;
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:215]:
```suggestion
```

Comment (nnyyxxxx) [inline core/tabs/applications-setup/browser-setup.sh:215]:
```suggestion
```

File: core/tabs/applications-setup/browser-setup.sh (hunk 6 above)
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:41]:
```suggestion
            "$ESCALATION_TOOL" curl http://dl.thorium.rocks/debian/dists/stable/thorium.list -o /etc/apt/sources.list.d/thorium.list
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/browser-setup.sh [hunk 10]
```diff
@@ -113,87 +122,100 @@ Signed-By: /usr/share/keyrings/librewolf.gpg" | "$ESCALATION_TOOL" tee /etc/apt/
...
-        *)
-            printf "%b\n" "${RED}The script does not support your Distro. Install manually..${RC}"
-            ;;
-    esac
+    if ! command_exists brave; then
+        case "$PACKAGER" in
+            apt-get|nala)
+                "$ESCALATION_TOOL" "$PACKAGER"  install curl
+                "$ESCALATION_TOOL" curl -fsSLo /usr/share/keyrings/brave-browser-archive-keyring.gpg https://brave-browser-apt-release.s3.brave.com/brave-browser-archive-keyring.gpg
+                echo "deb [signed-by=/usr/share/keyrings/brave-browser-archive-keyring.gpg] https://brave-browser-apt-release.s3.brave.com/ stable main"|sudo tee /etc/apt/sources.list.d/brave-browser-release.list
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/browser-setup.sh:137]:
```suggestion
                echo "deb [signed-by=/usr/share/keyrings/brave-browser-archive-keyring.gpg] https://brave-browser-apt-release.s3.brave.com/ stable main"| "$ESCALATION_TOOL" tee /etc/apt/sources.list.d/brave-browser-release.list
```

File: core/tabs/applications-setup/browser-setup.sh [hunk 11]
```diff
@@ -4,81 +4,90 @@
...
+        printf "%b\n" "${GREEN}Firefox Browser is already installed.${RC}"
+    fi
 }
 
 install_librewolf() {
 	printf "%b\n" "${YELLOW}Installing Librewolf...${RC}"
+    if ! command_exists librewolf; then
 	case "$PACKAGER" in
 		apt-get|nala)
 			"$ESCALATION_TOOL" "$PACKAGER" update && "$ESCALATION_TOOL" "$PACKAGER" install -y wget gnupg lsb-release apt-transport-https ca-certificates
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/browser-setup.sh:93]: use curl

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/browser-setup.sh [hunk 12]
```diff
@@ -4,84 +4,93 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" refresh
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install google-chrome-stable
+                ;;
+            pacman)
+                "$AUR_HELPER" -S --noconfirm google-chrome
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y fedora-workstation-repositories
+                "$ESCALATION_TOOL" "$PACKAGER" config-manager --set-enabled google-chrome
+                "$ESCALATION_TOOL" "$PACKAGER" install google-chrome-stable
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browser-setup.sh:24]:
```suggestion
                "$ESCALATION_TOOL" "$PACKAGER" install -y google-chrome-stable
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

**Review by ChrisTitusTech — CHANGES_REQUESTED** [review]
This is incorrect. Zypper requires --non-interactive and doesn't accept the -y flag.

**Review by ChrisTitusTech — APPROVED** [review]
nevermind tested in OpenSUSE.

--non-interactive or -n is valid if used before the install command
-y is valid if used AFTER the install command.

That is kind of insane both do the same thing.

**Pull request merged**